src/modules/UI/scenes/SendConfirmation/action.js

```diff
@@ -28,6 +39,55 @@ export const updateAmount = (nativeAmount: string, exchangeAmount: string, fiatP
   dispatch(createTX({ nativeAmount, metadata }, false))
 }
 
+type EdgePaymentProtocolUri = EdgeParsedUri & { paymentProtocolURL: string }
+
+export const paymentProtocolUriReceived = ({ paymentProtocolURL }: EdgePaymentProtocolUri) => (dispatch: Dispatch, getState: GetState) => {
+  const state = getState()
+  const walletId = getSelectedWalletId(state)
+  const edgeWallet = getWallet(state, walletId)
+
+  Promise.resolve(paymentProtocolURL)
+    .then(paymentProtocolURL => {
+      return getPaymentProtocolInfo(edgeWallet, paymentProtocolURL)
+    })
+    .then(paymentProtocolInfo => {
+      return makeSpendInfo(paymentProtocolInfo)
+    })
```

Comment:
And since `makeSpendInfo` is not an async function, then you can remove one of the `then` and just write it as:
.then(paymentProtocolInfo => {
  const spendInfo = makeSpendInfo(paymentProtocolInfo)
  dispatch(newSpendInfo(spendInfo))
  ...etc
 })

Reply:
once i enter async, i keep it async all the way, making each `.then` as small as possible, and using composition as much as possible

Reply:
i prefer smaller `.then`'s and composition